qiime2/core/archive/provenance_lib/_checksum_validator.py

```diff
@@ -104,31 +110,36 @@ def validate_checksums(
 
 def diff_checksums(zf: ZipFile) -> ChecksumDiff:
     '''
-    Calculates checksums for all files in an archive (except checksums.md5).
-    Compares these against the checksums stored in checksums.md5, returning
-    a summary ChecksumDiff.
+    Calculates checksums for all files in an archive
+    (except checksums.md5 or checksums.sha512,
+    depending on the archive version).
+    Compares these against the checksums stored in
+    checksums.md5/checksums.sha512, returning a summary ChecksumDiff.
 
     Parameters
     ----------
-    zf : ZipFile
+    zf : ZipFile\n
         The zipfile object of the archive.
 
     Returns
     -------
-    ChecksumDiff
+    ChecksumDiff\n
         A tuple of three dicts, one each for added, removed, and changed
         files. Keys are filepaths. For the added and removed dicts
         values are the checksum of the added or removed file. For the changed
         dict values are a tuple of (expected checksum, observed checksum).
+
     '''
     archive_version, _ = parse_version(zf)
     # TODO: don't think this is ever called
-    if int(archive_version) < 5:
+    if float(archive_version) < 5.0:
         return ChecksumDiff({}, {}, {})
 
+    checksum_ext = _parse_checksum_ext(zf)
+
     root_dir = pathlib.Path(get_root_uuid(zf))
-    checksum_fp = str(root_dir / 'checksums.md5')
-    obs = md5sum_directory_zip(zf)
+    checksum_fp = str(root_dir / f'checksums.{checksum_ext}')
+    obs = checksum_directory_zip(zf, checksum_type=checksum_ext)
```

Comment:
There's going to be an issue here, as the annotations are not exempted from the checksum list, so they will always look to be "added" files in the diff, and fail validation.

Reply:
ok this has been fixed - something that i'd also like to discuss though is why there were no test failures associated with this change

Reply:
evan and i discussed this offline, this is expected bc none of the tests include the addition of annotations, which is where we'd expect to see the difference (before this change was made). i am going to add a test to assert an empty diff for this change.